Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool/Config.hs

```diff
@@ -73,21 +84,59 @@ findDriverPoolConfig configs serviceTier tripCategory dist area = do
   find (filterByDistAndDvehAndArea serviceTier tripCategory dist area) configs
     <|> find (filterByDistAndDvehAndArea serviceTier tripCategory dist SL.Default) configs
 
+newtype Config = Config {config :: DriverPoolConfig} deriving (Generic, ToJSON, FromJSON, Show)
+
 getDriverPoolConfigFromDB ::
-  (CacheFlow m r, EsqDBFlow m r) =>
+  ( CacheFlow m r,
+    EsqDBFlow m r,
+    BeamFlow m r
+  ) =>
   Id MerchantOperatingCity ->
   Maybe DVST.ServiceTierType ->
   String ->
   SL.Area ->
   Maybe Meters ->
+  Maybe CacKey ->
   m (Maybe DriverPoolConfig)
-getDriverPoolConfigFromDB merchantOpCityId serviceTier tripCategory area mbDist = do
+getDriverPoolConfigFromDB merchantOpCityId serviceTier tripCategory area mbDist stickeyKey = do
   configs <- CDP.findAllByMerchantOpCityId merchantOpCityId
   transporterConfig <- CTC.findByMerchantOpCityId merchantOpCityId Nothing >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
   localTime <- getLocalCurrentTime transporterConfig.timeDiffFromUtc -- bounds, all these params, timeDiffFromUTC
   let boundedConfigs = findBoundedDomain (filter (\cfg -> cfg.timeBounds /= Unbounded) configs) localTime
   let unboundedConfig = filter (\cfg -> cfg.timeBounds == Unbounded) configs
-  return $
-    ( getDriverPoolConfigFromDB' serviceTier tripCategory area mbDist boundedConfigs
-        <|> getDriverPoolConfigFromDB' serviceTier tripCategory area mbDist unboundedConfig
-    )
+  let dpc' = getDriverPoolConfigFromDB' serviceTier tripCategory area mbDist boundedConfigs <|> getDriverPoolConfigFromDB' serviceTier tripCategory area mbDist unboundedConfig
+  oldVersion <- getConfigVersion (getKeyValue <$> stickeyKey)
+  (allLogics, version) <- TDL.getAppDynamicLogic merchantOpCityId LYT.CONFIG localTime oldVersion
+  case dpc' of
+    Just dpc -> do
+      let config = Config dpc
+      resp <- LYTU.runLogics allLogics config
+      case (fromJSON resp.result :: Result DriverPoolConfig) of
+        Success dpc'' -> do
+          Data.Foldable.forM_ version (cacheConfigVersion (getKeyValue <$> stickeyKey))
```

Comment:
can we set it only if it was not set earlier ?

Reply:
sure